osu.Game/Database/RealmExtensions.cs

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System.Collections.Generic;
+using AutoMapper;
+using osu.Game.Beatmaps;
+using osu.Game.Configuration;
+using osu.Game.Input.Bindings;
+using osu.Game.IO;
+using osu.Game.Rulesets;
+using osu.Game.Scoring;
+using osu.Game.Skinning;
+using Realms;
+
+namespace osu.Game.Database
+{
+    public static class RealmExtensions
+    {
+        private static readonly IMapper mapper = new MapperConfiguration(c =>
+        {
+            c.ShouldMapField = fi => false;
+            c.ShouldMapProperty = pi => pi.SetMethod != null && pi.SetMethod.IsPublic;
+
+            c.CreateMap<BeatmapDifficulty, BeatmapDifficulty>();
+            c.CreateMap<BeatmapInfo, BeatmapInfo>();
+            c.CreateMap<BeatmapMetadata, BeatmapMetadata>();
+            c.CreateMap<BeatmapSetFileInfo, BeatmapSetFileInfo>();
+
+            c.CreateMap<BeatmapSetInfo, BeatmapSetInfo>()
+             .ForMember(s => s.Beatmaps, d => d.MapFrom(s => s.Beatmaps))
+             .ForMember(s => s.Files, d => d.MapFrom(s => s.Files))
+             .MaxDepth(2);
+
+            c.CreateMap<DatabasedKeyBinding, DatabasedKeyBinding>();
+            c.CreateMap<RealmKeyBinding, RealmKeyBinding>();
+            c.CreateMap<DatabasedSetting, DatabasedSetting>();
+            c.CreateMap<FileInfo, FileInfo>();
+            c.CreateMap<ScoreFileInfo, ScoreFileInfo>();
+            c.CreateMap<SkinInfo, SkinInfo>();
+            c.CreateMap<RulesetInfo, RulesetInfo>();
```

Comment:
These are taken from my previous implementation attempt. Can remove if you'd prefer, but they will likely need to be added back in the future.